django_auth_adfs/backend.py

```diff
@@ -401,10 +401,9 @@ def update_user_groups(self, user, claim_groups):
                         for name in claim_groups
                         if name not in existing_claimed_group_names
                     ]
-                    # Associate the users to all claimed groups
-                    user.groups.set(
-                        tuple(existing_claimed_groups) + tuple(new_claimed_groups)
-                    )
+                    # Set user's groups to all claimed groups (both existing and
+                    # newly created) and remove any that are not in the claim.
+                    user.groups.set(new_claimed_groups)
```

Comment:
From my understanding, the comment here matches the old code. The new code will set the user's groups to only the groups in the claim that didn't already have a `Group` instance for them.